docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/2.deploy.md

```diff
@@ -0,0 +1,95 @@
+# Deploy full-text index
+
+Nebula Graph full-text indexes are powered by [Elasticserch](https://en.wikipedia.org/wiki/Elasticsearch). Full-text indexes are managed through built-in procedures. They can be created only for variable `STRING` and `FIXED_STRING` properties when the listener cluster and the elastic search cluster are ready.
+
+## Deploy Elasticsearch cluster
+
+To deploy an Elasticsearch cluster, see the [Elasticsearch documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-elasticsearch.html).
+
+When the Elasticsearch cluster is started, add the index template for the Nebula Graph full-text. For example:
+
+```json
+{
+ "template": "nebula*",
+  "settings": {
+    "index": {
+      "number_of_shards": 3,
+      "number_of_replicas": 1
+    }
+  },
+  "mappings": {
+    "properties" : {
+            "tag_id" : { "type" : "long" },
+            "column_id" : { "type" : "text" },
+            "value" :{ "type" : "text"}
+        }
+  }
+}
+```
+
+Make sure that you specify the following fields in strict accordance with the preceding template format:
+
+```text
+"template": "nebula*"
+"tag_id" : { "type" : "long" },
+"column_id" : { "type" : "text" },
+"value" :{ "type" : "text"}
+```
+
+## Deploy full-text cluster
+
+Full-text index data is wrote to the Elasticsearch cluster asynchronously. To use the full-text index search, you must run the full-text storage services as Listeners. Configure the `listener_path` parameter for the Listeners. For example:
+
+```conf
+########## basics ##########
+# The file to host the process id
+--pid_file=pids_listener/nebula-storaged.pid
+```
+
+Add the preceding configurations to the `nebula-graphd.conf` configuration file (the directory is `/usr/local/nebula/etc/` by default) for the Listeners.
+
+## Sign in and sign out to the Elasticsearch clients
+
+```ngql
+SIGN {IN | OUT} TEXT SERVICE [(<ip:port> [,<username>, <password>]), (<ip:port>), ...]
+```
+
+When the Elasticsearch cluster is deployed, use the `SIGN IN | OUT` statement to sign in or sing out to the Elasticsearch clients. Multiple `ip:port` pairs are separated with commas. You must use the real IPs. For example:
+
+```ngql
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200);
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200), (192.168.8.2:9300);
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200, "user", "password");
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200, "user", "password");, (127.0.0.1:9200, "user", "password");
+nebula> SIGN OUT TEXT SERVICE;
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200, "user");
+```
+
+## Add and remove full-text Listeners
+
+```ngql
+{ADD | REMOVE} LISTENER ELASTICSEARCH (<ip:port>) [,(<ip:port>), ...]
+```
```

Comment:
```
note > once you add a lisnter-ip to Nebula, it will begin to write to Elastic xxx. This may cause many reads 
from Nebula Storage,

 and many writes into ES. You may use the following paramter as flow-control.
```

Reply:
Don't understand. What is `the following parameter as flow-control`??

Reply:
> note > once you add a lisnter-ip to Nebula, it will begin to write to Elastic xxx. This may cause many reads 
> from Nebula Storage,
> 
>  and many writes into ES. You may use the following paramter as flow-control.

DEFINE_int32(listener_commit_interval_secs, 1, "Listener commit interval");
DEFINE_int32(listener_commit_batch_size, 1000, "Max batch size when listener commit");
DEFINE_int32(ft_request_retry_times, 3, "Retry times if fulltext request failed");
DEFINE_int32(ft_bulk_batch_size, 100, "Max batch size when bulk insert");